canvas_modules/common-canvas/src/common-properties/components/moveable-table-rows/moveable-table-rows.jsx

```diff
@@ -141,19 +158,49 @@ class MoveableTableRows extends React.Component {
 		return [topImages, bottomImages];
 	}
 
+	getLeastValue() {
+		let leastValue = 0;
+		const staticRows = this.props.controller.getStaticRows(this.props.propertyId).sort();
+		if (staticRows && staticRows.length > 0 && staticRows.includes(0)) {
+			leastValue = staticRows[staticRows.length - 1];
+		}
+		return leastValue;
+	}
+
+	getMaxValue() {
+		const controlValue = this.props.controller.getPropertyValue(this.props.propertyId);
+		let maxValue = controlValue.length - 1;
+		const staticRows = this.props.controller.getStaticRows(this.props.propertyId).sort();
+		if (staticRows && staticRows.length > 0 && staticRows.includes(controlValue.length - 1)) {
+			maxValue = staticRows[0] - 1;
+		}
+		return maxValue;
+	}
+
 	topMoveRow(evt) {
 		var selected = this.props.controller.getSelectedRows(this.props.propertyId).sort(function(a, b) {
 			return a - b;
 		});
 		const controlValue = this.props.controller.getPropertyValue(this.props.propertyId);
-		for (var firstRow = selected[0]; firstRow > 0; firstRow--) {
+		const leastValue = this.getLeastValue();
+		const staticRows = this.props.controller.getStaticRows(this.props.propertyId).sort();
+		for (var firstRow = selected[0]; firstRow > leastValue; firstRow--) {
 			for (var i = 0; i <= selected.length - 1; i++) {
-				const selectedRow = selected.shift();
-				const tmpRow = controlValue[selectedRow - 1];
-				controlValue[selectedRow - 1] = controlValue[selectedRow];
-				controlValue[selectedRow] = tmpRow;
-				selected.push(selectedRow - 1);
-				this.props.controller.moveErrorMessageRows(this.props.propertyId.name, selectedRow, selectedRow - 1);
+				if (staticRows.length > 0 && selected[0] > leastValue + 1) {
```

Comment:
These if cases have the same code in the block. Can you combine them into a single if statement?